src/package.cc

```diff
@@ -49,7 +51,13 @@ double Package::GetFillMass(double qty) {
       fill_mass = fill_max_;
     }
   }
-  return std::min(qty, fill_mass);
+  fill_mass = std::min(qty, fill_mass);
+  if (fill_mass >= fill_min_) {
```

Comment:
I think by returning this pair, you may not even need this test anymore, at all, because you've removed the pathological case.  It used to be possible to return `GetFillMass() = 0` because of the early return near the top of the function when `qty < fill_min_`.  You now return (0,0) for that case.  The only way we can get to this point in `GetFillMass()` is if `qty >= fill_min_`.  Therefore, we can guarantee that `fill_mass >= fill_min_` at this point and that `num_at_fill_mass` is well-behaved.